src/wp-includes/option.php

```diff
@@ -902,6 +911,24 @@ function update_option( $option, $value, $autoload = null ) {
 	return true;
 }
 
+/**
+ * Utility wrapper for update_option() that requires the old value to compare against
+ *
+ * @see update_option()
+ *
+ * @since 6.6.0
+ *
+ * @param string      $option    Name of the option to update. Expected to not be SQL-escaped.
+ * @param mixed       $value     Option value. Must be serializable if non-scalar. Expected to not be SQL-escaped.
+ * @param mixed       $old_value The option value retrieved with get_option() initially to compare against
+ *                               to avoid any intermittent changes to the option to update the option with an old value.
+ * @param string|bool $autoload  Optional. @see update_option(). Default null.
+ * @return bool True if the value was updated, false otherwise.
+ */
+function update_option_atomic( $option, $value, $old_value, $autoload = null ) {
```

Comment:
I'm -1 for adding such a new wrapper function, it's not consistent with other APIs like the metadata API for example.

I'd suggest focusing on the new param on `update_option` for now